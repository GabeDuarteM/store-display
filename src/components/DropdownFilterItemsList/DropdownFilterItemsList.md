```js
<div style={{ height: 155 }}>
  <DropdownFilterItemsList
    items={[
      { id: 'price-asc', text: 'Price: low to high' },
      { id: 'price-desc', text: 'Price: high to low' },
      { id: 'rating-asc', text: 'Rating: low to high' },
      { id: 'rating-desc', text: 'Rating: high to low' },
    ]}
    getItemProps={customProps => customProps}
    selectedItem={{ id: 'price-asc', text: 'Price: low to high' }}
    width={200}
  />
</div>
```
