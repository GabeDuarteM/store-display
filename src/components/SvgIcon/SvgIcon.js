import React from 'react'

import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import cn from 'classnames'

export const styles = {
  svgIcon: {
    height: ({ height }) => height,
    width: ({ width }) => width,
    fill: ({ color }) => color,
    display: 'block',
  },
}

// eslint-disable-next-line no-unused-vars
const SvgIcon = ({
  classes,
  src,
  height,
  width,
  color,
  className,
  ...rest
}) => (
  <span
    className={cn(classes.svgIcon, className)}
    // eslint-disable-next-line react/no-danger
    dangerouslySetInnerHTML={{ __html: src }}
    {...rest}
  />
)

export const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** If passed, the className will be applied to the root of the component */
  className: PropTypes.string,
  /** A string containing the content of the svg */
  src: PropTypes.string.isRequired,
  /** The height of the svg */
  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The width of the svg */
  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  /** The color to use as the fill of the svg */
  color: PropTypes.string,
}
export const defaultProps = {
  height: 'auto',
  width: 'auto',
  color: null,
  className: null,
}

SvgIcon.propTypes = propTypes
SvgIcon.defaultProps = defaultProps

export default injectSheet(styles)(SvgIcon)
