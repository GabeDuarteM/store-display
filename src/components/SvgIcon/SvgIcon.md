```js
const downArrow = require('../../assets/down-arrow.svg')
;<SvgIcon width={10} src={downArrow} />
```
