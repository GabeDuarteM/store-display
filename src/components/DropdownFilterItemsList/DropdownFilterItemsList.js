import React from 'react'

import PropTypes from 'prop-types'
import injectSheet from 'react-jss'

import checkMark from '../../assets/check-mark.svg'
import SvgIcon from '../SvgIcon'

export const styles = {
  itemList: {
    boxSizing: 'border-box',
    width: ({ width }) => width,
    position: 'absolute',
    marginTop: 4,
    cursor: 'pointer',
    border: '1px solid #e7e8ec',
    boxShadow: '0px 2px 6px rgba(115,115,115,0.12)',
    borderRadius: 3,
    backgroundColor: 'white',
    zIndex: 1,
  },
  item: {
    height: 36,
    padding: '0 16px',
    display: 'flex',
    alignItems: 'center',
    fontSize: 12,
    color: '#585858',
    '&:hover': {
      backgroundColor: '#f4f6f7',
    },
  },
  checkMark: {
    marginLeft: 'auto',
  },
}

const DropdownFilterItemsList = ({
  classes,
  items,
  getItemProps,
  selectedItem,
}) => (
  <div className={classes.itemList}>
    {items.map(item => (
      <div {...getItemProps({ item, className: classes.item })} key={item.id}>
        {item.text}
        {selectedItem.id === item.id && (
          <SvgIcon
            width={9}
            color="#6e8be3"
            src={checkMark}
            className={classes.checkMark}
          />
        )}
      </div>
    ))}
  </div>
)

const propTypes = {
  /** @ignore */
  classes: PropTypes.object.isRequired,
  /** Array of items to display */
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired,
    }),
  ).isRequired,
  /** Function to get the props from Downshift */
  getItemProps: PropTypes.func.isRequired,
  /** The current selected item to be highlighted with the check mark  */
  selectedItem: PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }).isRequired,
  /** The width of the component */
  width: PropTypes.number.isRequired, // eslint-disable-line react/no-unused-prop-types
}

DropdownFilterItemsList.propTypes = propTypes

export default injectSheet(styles)(DropdownFilterItemsList)
