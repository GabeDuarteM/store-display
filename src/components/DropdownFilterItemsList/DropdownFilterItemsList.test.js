import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import DropdownFilterItemsList, {
  styles,
  propTypes,
} from './DropdownFilterItemsList'
import SvgIcon from '../SvgIcon'

const createWrapper = (func, optionalProps) =>
  func(
    <DropdownFilterItemsList
      items={[
        { id: 'price-asc', text: 'Price: low to high' },
        { id: 'price-desc', text: 'Price: high to low' },
        { id: 'rating-asc', text: 'Rating: low to high' },
        { id: 'rating-desc', text: 'Rating: high to low' },
      ]}
      getItemProps={customProps => customProps}
      selectedItem={{ id: 'price-desc', text: 'Price: high to low' }}
      width={200}
      {...optionalProps}
    />,
  )

const createWrapperShallow = optionalProps =>
  createWrapper(shallow, optionalProps)

const createWrapperMount = optionalProps => createWrapper(mount, optionalProps)

describe('COMPONENT: <DropdownFilterItemsList />', () => {
  it('should render without crashing with default props', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it("should match component's style snapshot", () => {
    expect(styles).toMatchSnapshot()
  })

  it("should match component's propTypes snapshot", () => {
    expect(propTypes).toMatchSnapshot()
  })

  it('should spread the return of the method getItemProps on the root component', () => {
    const wrapper = createWrapperShallow()
    const { className } = wrapper.dive().props()
    expect(className).toBeDefined()
  })

  it('should render one item for each array item passed in the props', () => {
    const items = [
      { id: 'item-1', text: 'item 1' },
      { id: 'item-2', text: 'item 2' },
      { id: 'item-3', text: 'item 3' },
    ]
    const wrapper = createWrapperShallow({ items })
    const childrenLength = wrapper.dive().children().length
    expect(childrenLength).toBe(3)
  })

  it('should render each item displaying its text', () => {
    const items = [{ id: 'item-1', text: 'item 1' }]
    const wrapper = createWrapperShallow({ items })
    const text = wrapper
      .dive()
      .childAt(0)
      .text()
    expect(text).toBe(items[0].text)
  })

  it('should render an SvgIcon when the item is the selected item', () => {
    const selectedItem = { id: 'item-3', text: 'item 3' }
    const items = [
      { id: 'item-1', text: 'item 1' },
      { id: 'item-2', text: 'item 2' },
      selectedItem,
      { id: 'item-4', text: 'item 4' },
    ]
    const wrapper = createWrapperShallow({ items, selectedItem })
    const childrenLength = wrapper
      .dive()
      .childAt(2)
      .find(SvgIcon)
      .exists()
    expect(childrenLength).toBeTruthy()
  })
})
