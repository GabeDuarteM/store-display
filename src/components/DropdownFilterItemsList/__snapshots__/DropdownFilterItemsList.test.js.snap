// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <DropdownFilterItemsList /> should match component's propTypes snapshot 1`] = `undefined`;

exports[`COMPONENT: <DropdownFilterItemsList /> should match component's snapshot 1`] = `
<div
  className="DropdownFilterItemsList-itemList DropdownFilterItemsList-itemList"
>
  <div
    className="DropdownFilterItemsList-item DropdownFilterItemsList-item"
    item={
      Object {
        "id": "price-asc",
        "text": "Price: low to high",
      }
    }
    key="price-asc"
  >
    Price: low to high
  </div>
  <div
    className="DropdownFilterItemsList-item DropdownFilterItemsList-item"
    item={
      Object {
        "id": "price-desc",
        "text": "Price: high to low",
      }
    }
    key="price-desc"
  >
    Price: high to low
    <Jss(SvgIcon)
      className="DropdownFilterItemsList-checkMark DropdownFilterItemsList-checkMark"
      color="#6e8be3"
      height="auto"
      src=""
      width={9}
    />
  </div>
  <div
    className="DropdownFilterItemsList-item DropdownFilterItemsList-item"
    item={
      Object {
        "id": "rating-asc",
        "text": "Rating: low to high",
      }
    }
    key="rating-asc"
  >
    Rating: low to high
  </div>
  <div
    className="DropdownFilterItemsList-item DropdownFilterItemsList-item"
    item={
      Object {
        "id": "rating-desc",
        "text": "Rating: high to low",
      }
    }
    key="rating-desc"
  >
    Rating: high to low
  </div>
</div>
`;

exports[`COMPONENT: <DropdownFilterItemsList /> should match component's style snapshot 1`] = `
Object {
  "checkMark": Object {
    "marginLeft": "auto",
  },
  "item": Object {
    "&:hover": Object {
      "backgroundColor": "#f4f6f7",
    },
    "alignItems": "center",
    "color": "#585858",
    "display": "flex",
    "fontSize": 12,
    "height": 36,
    "padding": "0 16px",
  },
  "itemList": Object {
    "backgroundColor": "white",
    "border": "1px solid #e7e8ec",
    "borderRadius": 3,
    "boxShadow": "0px 2px 6px rgba(115,115,115,0.12)",
    "boxSizing": "border-box",
    "cursor": "pointer",
    "marginTop": 4,
    "position": "absolute",
    "width": [Function],
    "zIndex": 1,
  },
}
`;
